cypress/e2e/dashboard.cy.js

```diff
@@ -19,43 +19,74 @@ describe('Dashboard Application', () => {
+    })
+    it('has date filter custom range', () => {
+      cy.visit(`/a/${teamSlug}/dashboard/`)
+      cy.get('select[name="date_range"]').select('custom')
+      cy.get('input[name="start_date"]').should('be.visible')
+      cy.get('input[name="end_date"]').should('be.visible')
     })
   })
 
   describe('Dashboard Charts', () => {
-    it('displays session analytics chart', () => {
-      cy.visit(`/a/${teamSlug}/dashboard/`)
-      // Look for chart container (common class names for chart libraries)
-      cy.get('canvas, svg, .chart, .recharts-wrapper, .highcharts-container', { timeout: 10000 }).then(
-        ($chart) => {
-          if ($chart.length > 0) {
-            cy.log('Charts are rendered on dashboard')
-          }
-        }
-      )
-    })
+  it('overview statistics cards display', () => {
+    cy.visit(`/a/${teamSlug}/dashboard/`)
+
+    cy.get('.stat-card')
+      .should('have.length.greaterThan', 0)
+      .and('be.visible')
+
+    cy.get('.stat-card').should('have.length', 4)
+
+    cy.contains('.stat-label', 'Active Chatbots').should('exist')
+    cy.contains('.stat-label', 'Active Participants').should('exist')
+    cy.contains('.stat-label', 'Completed Sessions').should('exist')
+    cy.contains('.stat-label', 'Total Messages').should('exist')
+  })
 
     it('message volume chart loads', () => {
-      cy.visit(`/a/${teamSlug}/dashboard/`)
-      cy.contains(/Message|Volume|Traffic/i).then(($section) => {
-        if ($section.length > 0) {
-          cy.log('Message volume section found')
-        }
+        cy.visit(`/a/${teamSlug}/dashboard/`)
+
+        cy.contains(/Message|Volume|Traffic/i)
+          .should('exist')
+          .and('be.visible')
+
+        cy.get('#messageVolumeChart')
+              .should('exist')
+              .and('be.visible')
       })
-    })
 
     it('bot performance metrics display', () => {
       cy.visit(`/a/${teamSlug}/dashboard/`)
-      cy.contains(/Performance|Bot|Metrics/i).then(($section) => {
-        if ($section.length > 0) {
-          cy.log('Performance metrics section found')
-        }
-      })
+
+      // Assert the table exists
+      cy.get('#botPerformanceTable')
+        .should('exist')
+        .and('be.visible')
+
+      // Verify table has headers
+      cy.get('#botPerformanceTable thead th')
+        .should('have.length.greaterThan', 0)
+
+      // Verify table has at least one row of data
+      cy.get('#botPerformanceTable tbody tr')
+        .should('have.length.greaterThan', 0)
+
+      // Verify the first row contains actual data (chatbot name link)
+      cy.get('#botPerformanceTable tbody tr')
+        .first()
+        .find('a.btn')
+        .should('exist')
+        .and('have.attr', 'href')
+
+      // Verify numeric columns contain data
+      cy.get('#botPerformanceTable tbody tr')
+        .first()
+        .find('td')
+        .eq(1) // Participants column
+        .invoke('text')
+        .should('match', /\d+/)
```

Comment:
This doesn't seem quite correct since participant IDs can contain characters too. We can probably just remove this test
```suggestion
```